src/javascripts/components/search.js

```diff
@@ -0,0 +1,94 @@
+/* global XMLHttpRequest */
+import accessibleAutocomplete from 'accessible-autocomplete'
+import lunr from 'lunr'
+
+var searchIndex = null
+var documentStore = null
+var timeout = 5 // Time to wait before giving up fetching the search index
+var search = {
+  fetchSearchIndex: function (callback) {
+    var request = new XMLHttpRequest()
+    request.open('GET', '/search-index.json', true)
+    request.timeout = timeout * 1000
+
+    request.onload = function () {
+      if (request.status >= 200 && request.status < 400) {
+        // Success!
+        var response = request.responseText
+        var json = JSON.parse(response)
+        callback(json)
+      } else {
+        // We reached our target server, but it returned an error
+        console.error('onload error', request.status, request.responseText)
+        // Log to analytics?
+      }
+    }
+
+    request.onerror = function (error) {
+      // There was a connection error of some sort
+      console.error('onerror', error)
+      // Log to analytics?
+    }
+
+    request.ontimeout = function (error) {
+      console.error('ontimeout', error)
+      // Log timeout to analytics?
+    }
+
+    request.send()
+  },
+  searchEngine: function (query, callback) {
+    var matchedResults = []
+    if (!searchIndex || !documentStore) {
+      return callback(matchedResults)
+    }
+    var searchResults = searchIndex.query(function (q) {
+      q.term(query, { wildcard: lunr.Query.wildcard.TRAILING })
+    })
+    matchedResults = searchResults.map(function (result) {
+      return documentStore[result.ref]
+    })
+    console.log(matchedResults)
+    callback(matchedResults)
+  },
+  handleOnConfirm: function (result) {
+    var path = result.path
+    if (path) {
+      window.location.pathname = path
+    }
+  },
+  inputValueTemplate: function (result) {
+    return result && result.title
```

Comment:
It's not entirely clear what this is doing – is the check of `result` necessary? Will accessible-autocomplete ever call this with a falsey result?

Reply:
accessible autocomplete requires results and then results.key

Reply:
this will be called falsy if there are no results, for example if you search for something that doesnt exist